components/server/src/api/server.ts

```diff
@@ -211,16 +242,49 @@ export class API {
         };
     }
 
-    private async verify(context: HandlerContext) {
-        const user = await this.sessionHandler.verify(context.requestHeader.get("cookie") || "");
-        if (!user) {
+    private async verify(context: HandlerContext): Promise<string> {
+        const claims = await this.sessionHandler.verifyJWTCookie(context.requestHeader.get("cookie") || "");
+        const subjectId = claims?.sub;
+        if (!subjectId) {
             throw new ConnectError("unauthenticated", Code.Unauthenticated);
         }
-        const fgaChecksEnabled = await isFgaChecksEnabled(user.id);
+        return subjectId;
+    }
+
+    private async ensureFgaMigration(userId: string): Promise<User> {
+        const fgaChecksEnabled = await isFgaChecksEnabled(userId);
```

Comment:
should check for writes enabled

Reply:
sorry, I was reading the code wrong. Forgot we do check and disable public api all together when FGA is disabled.